### PR TITLE
Offers fixes

### PR DESCRIPTION
**Pull request by rustyrussell**
Mainly due to great feedback from @shesek !

### CODE REVIEW

**Review by vincenzopalazzo — APPROVED** [review]
ack https://github.com/ElementsProject/lightning/pull/4669/commits/2f0299448628b3077d635a968c5a1fb4d7c54767

Wow, enough code to digest for today, I hope I didn't miss nothings :)

**Comment by shesek** [comment]
Thanks for the swift fixes! This seems to cover everything that I reported, including #4666 and #4667.

File: plugins/offers_offer.c [hunk 1]
```diff
@@ -255,14 +255,39 @@ struct offer_info {
...
+					   void *arg UNNEEDED)
+{
+	bool active;
+
+	/* If it's inactive, we can't return it, */
+	if (!json_to_bool(buf, json_get_member(buf, result, "active"),
+			  &active)) {
+		return command_fail(cmd,
+				    LIGHTNINGD,
+				    "Bad creaoffer status reply %.*s",
```
Comment (niftynei) [inline plugins/offers_offer.c:270]: `creaoffer` -> `createoffer`

**Review by niftynei — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/4669/commits/2f0299448628b3077d635a968c5a1fb4d7c54767

Spelling nit in an error msg, but not worth holding the release for it imo

**Pull request merged**